### PR TITLE
KAROO-9382: Add eBike Summary Data

### PR DESCRIPTION
**Pull request by petercooke536**
- Add eBike option fields EbikeBatteryLevel, EbikeAssistMode, EbikeAssistLevelPercent
- create extraProductFields map in `sdk.go` to easily allow adding option fields

### CODE REVIEW

File: cmd/fitgen/internal/profile/transform.go [hunk 1]
```diff
@@ -162,7 +162,8 @@ func (f *Field) setLength() {
 
 func (f *Field) transform(subfield bool, ftypes map[string]*Type, handleHRSTQuirk bool, logger *log.Logger) (skip bool, err error) {
 	if f.data[mEXAMPLE] == "" || f.data[mEXAMPLE] == "0" {
-		if f.data[mFNAME] == "heart_rate_source_type" && handleHRSTQuirk {
+		_, ok := extraProductFields[f.data[mFNAME]]
```
Comment (cjstoltzfus) [inline cmd/fitgen/internal/profile/transform.go:165]: Can you elaborate more on why these changes were necessary? Do the ebike fields suffer from the same problem as the HRST field in the FIT SDK spreadsheet? Also, you're no longer checking the `handleHRSTQuirk` flag, which might be fine assuming it's always true.

Comment (petercooke536) [reply cmd/fitgen/internal/profile/transform.go:165]: These were optional and are not present in the RecordMsg in service so I could not properly access them.  The `handleHRSTQuirk` flag seemed pointless since I always had to build using the flag to make it true and it was not very straight forward at first.

Comment (cjstoltzfus) [reply cmd/fitgen/internal/profile/transform.go:165]: What do you mean by optional? Basically every field in a `RecordMsg` is optional.

Comment (petercooke536) [reply cmd/fitgen/internal/profile/transform.go:165]: I guess more like optionally complied but they were not in the RecordMsg.  We tried first upgrading out FitSDK to the latest but it was still not include and it started removing some fields which is why we took this route.   @dudanian Could probably better explain since he came up with this solution.

Comment (petercooke536) [reply cmd/fitgen/internal/profile/transform.go:165]: Optional is definitely not the right word, but it is probably suffering the same issue as the HRST

Comment (cjstoltzfus) [reply cmd/fitgen/internal/profile/transform.go:165]: Weird, ok. Guess we'll roll with it.

**Review by cjstoltzfus — APPROVED** [review]
